_posts/2021-04-08-pulp-3.12-is-generally-available.md

```diff
@@ -0,0 +1,119 @@
+
+A new warning has been added to the on-demand sync process to check if remote artifacts are detected with checksums that do not correspond with the `ALLOWED_CONTENT_CHECKSUMS`.
+
+### New sync check for forbidden checksums [#8423](https://pulp.plan.io/issues/8423)
+
+During a sync of on-demand content, a new check is performed and throws an error if forbidden checksums are detected.
+
+As part of [#7989](https://pulp.plan.io/issues/7989) more enhancements updates were made to handle the exception that gets raised on the client side.
+
+## Increased flexibility with deleting repository versions [#8454](https://pulp.plan.io/issues/8454)
+
+With this release, it is now possible to delete the original repository version (`RepoVersion 0`). As long as you have more than one repository version, you can now delete `RepoVersion 0`.
+
+## Breaking changes and behavior changes
+
+### PulpExporter: Update and Delete become asynchronous [#7438](https://pulp.plan.io/issues/7438)
+
+The Update and Delete endpoints of `PulpExporter` have been changed to become asynchronous. This change provides proper locking on resources.
+
+Note that this is a breaking change.
+
+### Deprecation warnings added to log level WARNING. [#8499 ](https://pulp.plan.io/issues/8499)
+
+In previous versions of Pulp, no one could see deprecation warnings for features. With this release, if the log level is set to `WARNING`, you'll now find deprecation warnings included in your logs.
+
+You can disable this by adjusting the log level of `pulpcore.deprecation`.
+
+## Updates to the Plugin API
+
+This release includes the following updates to the Plugin API.
+
+### New callback method for autopublishing [#7626](https://pulp.plan.io/issues/7626)
+
+A new callback method ``on_new_version()`` has been added to ``Repository``. This runs when a new repository version has been created. You can use this method for automatically publishing or distributing the new repository version after it has been created.
+
+### New logging options for ``DigestValidationError`` and ``SizeValidationError`` [#8357](https://pulp.plan.io/issues/8357)
+
+A `url` option has been added as a possible argument to ``DigestValidationError`` and ``SizeValidationError`` exceptions, so that you can better locate the url of the requested files.
+
+### A new Distribution MasterModel [#8384](https://pulp.plan.io/issues/8384)
+
+This release provides a new MasterModel `pulpcore.plugin.models.Distribution` that replaces the BaseDistribution `pulpcore.plugin.models.BaseDistribution` model.
+You will have to define a migration to move your data from `pulpcore.plugin.models.BaseDistribution` to `pulpcore.plugin.models.Distribution`. See the `pulp_file` migration 0009 as a reference example.[#8385](https://pulp.plan.io/issues/8385)
+
+This new model contains the `repository`, `repository_version`, and `publication` fields  instead of on the detail models as was done with `pulpcore.plugin.models.BaseDistribution`.
```

Comment:
```suggestion
This new model now contains the `repository`, `repository_version`, and `publication` fields that are supposed to move off of the detail models where they are located with `pulpcore.plugin.models.BaseDistribution`.
```

Reply:
@mdellweg what do you mean by supposed to? Does it or doesn't it?

Reply:
I added your sentence although I don't know what it means!

Reply:
The plugin migration you are mentioning before is supposed to move those fields.

Reply:
Ah I understand this now. Are we good to go with this section at least?